### DISCUSSION TITLE
NoTrainData Error when using the Generic Dataset

### DISCUSSION THREAD

**Discussion by mrosati20** [neuralhydrology/neuralhydrology#193] [Q&A] [answered]
Hello everyone!
I’ve been having a bit of an issue using the generic dataset. The output log says there isn’t a single valid sample in any of the basins during the training period. I’ve double checked that the files contain data for the period and that the date formats match but the error seems to persist no matter what I try. Does anyone have an idea of what I might be doing wrong?

Here are the original csv files used, the script used to convert the data to .nc format, the output log, the config file, and the run file I’ve been using:

[Arroyo_seco.txt](https://github.com/user-attachments/files/16314441/Arroyo_seco.txt)
[Arroyo_Seco_3_Basin.txt](https://github.com/user-attachments/files/16314442/Arroyo_Seco_3_Basin.txt)
[Hondo.csv](https://github.com/user-attachments/files/16314444/Hondo.csv)
[Lucero.csv](https://github.com/user-attachments/files/16314446/Lucero.csv)
[nc_conversion.txt](https://github.com/user-attachments/files/16314510/nc_conversion.txt)
[output.log](https://github.com/user-attachments/files/16314448/output.log)
[seco_run.txt](https://github.com/user-attachments/files/16314449/seco_run.txt)
[Taos.csv](https://github.com/user-attachments/files/16314450/Taos.csv)
[attributes.csv](https://github.com/user-attachments/files/16314481/attributes.csv)

**Comment by gauchm** [comment, marked as answer]
The only thing I see that seems suspicious: I think that swe(mm) is zero everywhere, so the 0-mean unit-variance scaling might turn it into NaNs. So I'd try removing swemm from the dynamic inputs. But I'm a bit surprised---I thought we had a separate error for that.
If that doesn't help, I'd try to pin down the problem: try not using static attributes, try removing individual dynamic attributes, etc., and see if any of those make it work.

**Reply by mrosati20** [reply]
Thanks! Removing the swe(mm) column from the data seems to have solved the problem.

**Reply by gauchm** [reply]
Great, thanks for reporting back, and glad we solved it. Just for the sake of completeness:

> I thought we had a separate error for that.

The special error I was thinking of is for static attributes, not dynamic ones. So this is as expected.
